llvm/include/llvm/BinaryFormat/DXContainer.h

```diff
@@ -546,13 +560,75 @@ struct ProgramSignatureElement {
 static_assert(sizeof(ProgramSignatureElement) == 32,
               "ProgramSignatureElement is misaligned");
 
+struct RootConstants {
+  uint32_t ShaderRegister = 0;
+  uint32_t RegisterSpace = 0;
+  uint32_t Num32BitValues = 0;
+
+  RootConstants() = default;
```

Comment:
Why does this have a constructor? We generally don't have constructors for types that are encoded in binary files because the common usage is to just cast a pointer and read.